src/cargo/util/toml_mut/manifest.rs

```diff
@@ -517,3 +539,14 @@ fn parse_manifest_err() -> anyhow::Error {
 fn non_existent_table_err(table: impl std::fmt::Display) -> anyhow::Error {
     anyhow::format_err!("the table `{table}` could not be found.")
 }
+
+fn non_existent_dependency_err(
+    name: impl std::fmt::Display,
+    table: impl std::fmt::Display,
+) -> anyhow::Error {
+    anyhow::format_err!(
+        "The dependency `{}` could not be found in `{}`.",
```

Comment:
side note: A future improvement here is to suggest a dependency with similar name to remove. 

```
The dependency `cat` could not be found in `dependencies`.
"Did you mean `bat` in `dev-dependencies`.
```

(mind-exploding)